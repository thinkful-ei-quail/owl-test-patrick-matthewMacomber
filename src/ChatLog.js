import React from 'react';
import store from './Store';

function findUser(id) {
  return store.find(user => user.id === id);
}

function printMessage(chatObj) {
  const user = findUser(chatObj.participantId);
  let message = '';
  switch(chatObj.type) {
    case 'message':
      return <div key={chatObj.timestamp}>
      <div>
      <img src={user.avatar} alt="avatar"/>
      <h3>${user.name}</h3>
      {new Date(chatObj.timestamp).toLocaleTimeString('en-US')}
    </div>
      {chatObj.message}
    </div>;
    case 'thumbs-up':
      message = `${user.name} gave a thumbs up &#128077;`;
    break;
    case 'thumbs-down':
      message = `${user.name} gave a thumbs down &#128078;`;
      break;
    case 'raise-hand':
      message = `${user.name} raised their hand &#9995;`;
      break;
    case 'clap':
      message = `${user.name} clapped &#128079;`;
      break;
    case 'join':
      message = `${user.name} joined`;
      break;
    case 'leave':
      message = `${user.name} left`;
      break;
    case 'join-stage':
      message = `${user.name} joined the stage`;
      break;
    case 'leave-stage':
      message = `${user.name} left the stage`;
      break;
    default :
      return '';
  }
  return <div key={chatObj.timestamp}>
    {message}
    </div>;
}

function ChatLog(props) {

  return <section className="chat-log">
    {props.messages.map(message => {
      return printMessage(message);
    })}
  </section>;
}

ChatLog.defaultProps = {
  messages: []
};

export default ChatLog;