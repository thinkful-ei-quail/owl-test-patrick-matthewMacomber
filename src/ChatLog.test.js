import React from 'react';
import ChatLog from './ChatLog';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import messages from './ChatMessages';

describe('Testing ChatLog component', () => {
  it('Render component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatLog messages={messages}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('Render component without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatLog />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatLog messages={messages}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});