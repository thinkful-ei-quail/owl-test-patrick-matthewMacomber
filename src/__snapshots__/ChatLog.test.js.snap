// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ChatLog component renders the UI as expected 1`] = `
<section
  className="chat-log"
>
  <div>
    <div>
      <img
        alt="avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3>
        $
        Axe Kubicka
      </h3>
      7:48:04 AM
    </div>
    Hello world
  </div>
  <div>
    Caty Flucker gave a thumbs up &#128077;
  </div>
  <div>
    Frank Runciman gave a thumbs down &#128078;
  </div>
  <div>
    Koren Templeton raised their hand &#9995;
  </div>
  <div>
    Koren Templeton clapped &#128079;
  </div>
  <div>
    Koren Templeton joined
  </div>
  <div>
    Ashla Attwool left
  </div>
  <div>
    Axe Kubicka joined the stage
  </div>
  <div>
    Axe Kubicka left the stage
  </div>
</section>
`;
